### DISCUSSION TITLE
Error handling in Go SDK

### DISCUSSION THREAD

**Discussion by cellz-me** [pulumi/pulumi#7306] [Pulumi in Practice] [answered]
Question: is there any downside to simply calling `panic()` when an error occurs (for instance when a resource update fails)? 

The alternative (which has been my approach so far) is to return the error up the chain of functions until eventually you get back to the function called by `pulumi.Run()`. This means a bunch of functions need to return two values. 

I have no problem with this if there is a good reason for it. But if calling `panic` works just as well, why not make the code a little simpler? Or am I missing something?

**Comment by lukehoban** [comment, marked as answer]
The approach of returning `error`s is idiomatic for Go, and also ensures that your program cooperatively cleans up as it exits.   It is definitely recommended to return `errors` if you can, though in general things should work if you `panic`.
